discovery/discovery.go

```diff
@@ -15,6 +16,11 @@ type Entry struct {
 	Port string
 }
 
+// TLS is exported
+type TLS struct {
```

Comment:
do we still need the type TLS struct ? I think we can just use Config *tls.Config